src/index.ts

```diff
@@ -156,6 +158,21 @@ const matchJsonToSchema = (
   indexes: number[] = []
 ): boolean => {
   try {
+    if (typeof schema === 'object' && schema['$not']) {
+      const result = matchJsonToSchema(
+        input,
+        schema['$not'],
+        root_input,
+        indexes
+      );
+
+      delete schema['$not'];
+
+      if (Object.keys(schema).length === 0 || result) {
```

Comment:
I'm not sure what this length check is useful for?

Reply:
Here we need to return the result if there is no more schema to validate because if the input is a primitive we don't get the expected result.

Let's say we take the structure of a test, [input, schema, match]:

```
[1, {}, false] // this is the actual behavior expected
```
But let's say we do this:
```
[1, { $not: 2 }, true]
```
If we don't stop validating after the $not then the condition becomes:
```
[1, {}, true]
```
Which will always evaluate as false even if we expect true.